src/vs/base/common/map.ts

```diff
@@ -314,7 +373,11 @@ export class TernarySearchTree<K, V> {
 				break;
 			}
 		}
-		return node ? node.value : undefined;
+		return node;
+	}
+
+	has(key: K): boolean {
```

Comment:
Actually unused and those changes are not needed?

Reply:
`has` is used here: https://github.com/microsoft/vscode/blob/54616132915d13c6a1d3cc864c2c64cba84c49cd/src%2Fvs%2Fplatform%2Fcontextkey%2Fbrowser%2FcontextKeyService.ts#L125

And I added that to ensure behavior wouldn't change, because there wasn't a way to test for the existence of a key -- because it could have a falsy value.

Reply:
makes sesne